test/library/standard/Path/dlongnecke/normPath/normPath.chpl

```diff
@@ -0,0 +1,74 @@
+/*
+   Collapse paths such as `foo//bar`, `foo/bar/`, `foo/./bar`, and
+   `foo/baz/../bar` into `foo/bar`.  Warning: may alter meaning of paths
+   containing symbolic links.
+
+   .. note::
+
+      This is currently only implemented in a Unix environment.  It will not
+      behave correctly in a non-Unix environment.
+
+   :arg name: a potential path to collapse, possibly destroying the meaning of
+              the path if symbolic links were included.
+   :type name: `string`
+
+   :return: the collapsed version of `name`
+   :rtype: `string`
+*/
+
+
+use Path;
+
+
+// As per the examples given in the description of the normPath function.
+const collapsedA = 'foo/bar';
+const collapseToA0 = 'foo//bar';
+const collapseToA1 = 'foo/bar/';
+const collapseToA2 = 'foo/./bar';
+const collapseToA3 = 'foo/baz/../bar';
+
+// Variables for expanded coverage.
+const multiCurDir = '././.';
+const multiParentDir = '../../..';
+const multiMixedParentCurDir = './.././.././..';
+const multiOnlyPathSep = '/////';
+const pathSep2 = '//';
```

Comment:
These are great things to check!  Some other cases that occurred to me are:
- "/.."
- "/."
- "foo/bar/."
- "foo/bar/baz/.."